source/run-command.txt

```diff
@@ -0,0 +1,182 @@
+    :end-before: end-execute-command
+
+For a full list of database commands and corresponding parameters, see the
+:ref:`Additional Information <c-addl-info-runcommand>` section.
+
+Command Options
+---------------
+
+You can specify optional command behavior for the
+``mongoc_client_command_with_opts()`` method.
+
+The ``mongoc_client_command_with_opts()`` method accepts a BSON document for the
+``opts`` parameter. This can include command options such as read concerns,
+write concerns, and session options. 
+
+You can pass a BSON document with the following options:
+
+- ``readConcern``
+- ``writeConcern``
+- ``sessionId``
+- ``collation``
+- ``serverId``
+
+To learn more about a command and the options that it accepts, locate the
+command and follow the link in the :manual:`Database Commands
+</reference/command/>` section of the Server manual.
+
+The following code shows you how to specify a ``grantRolesToUsers`` command with a
+``majority`` write concern:
+
+.. literalinclude:: /includes/usage-examples/run-command.c
+    :language: c
+    :dedent:
+    :start-after: start-execute-command-with-options
+    :end-before: end-execute-command-with-options
+
+.. note:: Read Preference
+
+   The ``mongoc_client_command_simple()`` and
+   ``mongoc_client_command_with_opts()`` methods ignore the read preference
+   setting you may have set on your ``Db`` object. By default, these methods use
+   the ``primary`` read preference.
+
+   To specify a read preference other than the primary read preference, you need
+   to explicity pass it as an argument. The following code demonstrates how to
+   specify a read preference and use it with the
+   ``mongoc_client_command_simple()`` method:
+
+   .. code-block:: C
+
+      read_prefs = mongoc_read_prefs_new(MONGOC_READ_SECONDARY);
+
+      command = BCON_NEW("hello", BCON_INT32(1));
+
+      retval = mongoc_client_command_simple(client, "admin", command,
+      read_prefs, NULL, &reply, &error);
+  
+   For more information on read preference options, see :manual:`Read Preference
+   </core/read-preference/>` in the Server manual.
+
+Response
+--------
+
+Each method returns either a BSON document or a cursor containing the response
+from the database after the command runs. Each database command performs a
+different function, so the response content can vary across commands. However,
+every response contains documents with the following fields:
+
+- **<command result>:** Provides fields specific to the database command. For
+  example, ``count`` returns the ``n`` field and ``explain`` returns the
+  ``queryPlanner`` field.
+- ``ok``: Indicates whether the command succeeded (``1``) or failed (``0``).
+
+Example
+-------
+
+The following code shows how you can use the
+``mongoc_client_command_with_opts()`` method to run the
+``cloneCollectionAsCapped`` command with the ``writeConcern`` option, then run
+the ``distinct`` command with the ``collation`` and ``readConcern`` options.
```

Comment:
Since this example is so long, can we highlight the relevant bits? At least the methods/options that are called out in this paragraph

Reply:
Makes sense! I added the lines to emphasize but not seeing the highlighting in the preview, lmk if you can tell what the include is missing 😅

Reply:
It looks like you added the actual line numbers from the file, but I think the `:emphasize-lines:` is relative to the included code block, not the actual file. So the first line that gets included would be line 1, regardless of what line it is in the file

Reply:
Ah ok that makes sense! Thank you 🙏